File: ruby/011.rb
# 011
# Container With Most Water
# 2014-04-19
#####################################################
# @param {Integer[]} height
# @return {Integer}
def max_area(height)
  i = 0
  j = height.size - 1
  maxA = 0
  while i < j
    maxA = [maxA, (j-i) * [height[i], height[j]].min].max
    height[i] < height[j] ? i+=1 : j-=1
  end
  maxA
end

#####################################################

height = [1, 2, 4]
puts max_area(height)

File: python/011.py
#   011
#   Container With Most Water
#   2014-04-19
#####################################################
class Solution:
    # @param {integer[]} height
    # @return {integer}
    def maxArea(self, height):
        i = 0
        j = len(height) - 1
        maxA = 0
        while i < j:
            maxA = max(maxA, (j-i) * min(height[i], height[j]))
            if height[i] < height[j]:
                i+=1
            else:
                j-=1
        return maxA

#####################################################

height = [1, 2, 4]
print(height)
sol = Solution()
z = sol.maxArea(height)
print(z)

